### PR TITLE
✨[RUM-5090] Collect ressource protocol

### PR DESCRIPTION
**Pull request by RomanGaignault**
## Motivation

Adding the network protocol information (e.g., http/1.1, h2) to resource events will provide deeper insights into how resources are fetched by the browser. 

## Changes

Utilized performanceTiming.nextHopProtocol to obtain the protocol information for same-origin requests.
Note: nextHopProtocol is only available for same-origin resources and might be restricted by the Timing-Allow-Origin header for cross-origin resources.

## Testing

- [x] Local
- [x] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

**Review by BenoitZugmeyer — COMMENTED** [review]
You'll need to add that field in rum-events-format (around [here](https://github.com/DataDog/rum-events-format/blob/ba50d0c05dcdd374bb962f40fc8f82db992da319/schemas/rum/resource-schema.json#L215)), get it merged, run `rum-events-format:sync` in the SDK repo and update this PR

**Review by BenoitZugmeyer — APPROVED** [review]

**Review by BenoitZugmeyer — COMMENTED** [review]
Put it in staging before merging the PR, to make sure it works as expected there.

File: packages/rum-core/src/domain/requestCollection.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ export interface RequestCompleteEvent {
   method: string
   url: string
   status: number
+  protocol?: string
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/requestCollection.ts:47]: **💬 suggestion:** ‏Remove this property as it is never defined

File: packages/rum-core/src/domain/resource/resourceCollection.ts [hunk 2]
```diff
@@ -232,3 +234,12 @@ function computeRequestDuration(pageStateHistory: PageStateHistory, startClocks:
...
 }
+
+/**
+ * The 'nextHopProtocol' is an empty string for cross-origin resources without CORS headers,
+ * meaning the protocol is unknown, and we shouldn't report it.
+ * https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming/nextHopProtocol#cross-origin_resources
+ */
+function discardEmptyProtocol(requestProtocol: string | undefined): string | undefined {
+  return requestProtocol === '' ? undefined : requestProtocol
+}
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/resource/resourceCollection.ts:245]: **💬 suggestion:** to refactor things a bit further, ‏move this function to `resourceUtils.ts` and change it like this:

```ts
export function computeResourceEntryProtocol(entry: RumPerformanceResourceTiming) {
  return entry.nextHopProtocol === '' ? undefined : entry.nextHopProtocol
}
```

File: packages/rum-core/src/domain/resource/resourceCollection.ts [hunk 3]
```diff
@@ -104,6 +104,7 @@ function processRequest(
         duration,
         method: request.method,
         status_code: request.status,
+        protocol: request.protocol,
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/resource/resourceCollection.ts:107]: **💬 suggestion:** ‏
```suggestion
        protocol: matchingTiming && computeResourceEntryProtocol(matchingTiming)
```

**Review by BenoitZugmeyer — APPROVED** [review]

**Review by cy-moi — APPROVED** [review]
Nice!

**Pull request merged**